src/drivers/azure-storage-blob.ts

```diff
@@ -24,41 +24,69 @@ export interface AzureStorageBlobOptions {
   accountKey?: string;
 
   /**
-   * The SAS key. If provided, the account key will be ignored.
+   * The SAS token. If provided, the account key will be ignored. Include at least read, list and write permissions to be able to list keys.
    */
   sasKey?: string;
 
+  /**
+   * The SAS URL. If provided, the account key, SAS key and container name will be ignored.
+   */
+  sasUrl?: string;
+
   /**
    * The connection string. If provided, the account key and SAS key will be ignored. Only available in Node.js runtime.
    */
   connectionString?: string;
+
+  /**
+   * Storage account endpoint suffix. Need to be changed for Microsoft Azure operated by 21Vianet, Azure Government or Azurite.
+   * @default ".blob.core.windows.net"
+   */
+  endpointSuffix?: string;
 }
 
 const DRIVER_NAME = "azure-storage-blob";
 
 export default defineDriver((opts: AzureStorageBlobOptions) => {
   let containerClient: ContainerClient;
+  const endpointSuffix = opts.endpointSuffix || ".blob.core.windows.net";
   const getContainerClient = () => {
     if (containerClient) {
       return containerClient;
     }
-    if (!opts.accountName) {
-      throw createError(DRIVER_NAME, "accountName");
+    if (!opts.connectionString && !opts.sasUrl && !opts.accountName) {
+      throw createError(DRIVER_NAME, "missing accountName");
     }
     let serviceClient: BlobServiceClient;
     if (opts.accountKey) {
       // StorageSharedKeyCredential is only available in Node.js runtime, not in browsers
       const credential = new StorageSharedKeyCredential(
-        opts.accountName,
+        opts.accountName!,
```

Comment:
do you need this non-null assertion anymore?

```suggestion
         opts.accountName,
```

Reply:
This is from my initial implementation. I tried to incorporate as many combinations as possible, as not all of the combinations are mutually exclusive.